config/configgrpc/configgrpc.go

```diff
@@ -270,7 +270,7 @@ func (gss *GRPCServerSettings) ToListener() (net.Listener, error) {
 }
 
 // ToServerOption maps configgrpc.GRPCServerSettings to a slice of server options for gRPC.
-func (gss *GRPCServerSettings) ToServerOption(ext map[config.ComponentID]component.Extension) ([]grpc.ServerOption, error) {
+func (gss *GRPCServerSettings) ToServerOption(ext map[config.ComponentID]component.Extension, settings component.ReceiverCreateSettings) ([]grpc.ServerOption, error) {
```

Comment:
This makes an assumption that `GRPCServerSettings` are only used for "receivers". I thought about this before and what I have in my mind is to extract {Logger, TP, MP} into a "TelemetryCreateSettings" struct that is included in all the components CreateSettings, and in cases like this pass only the TelemetryCreateSettings. Something like this, which will be minimal breaking change, since the members can still be accessed as before, only the construction of the settings will change which probably should always go via the helpers.

If you agree with this change we should have 2 PRs: 1. makes the split of structs; 2. adds the MP.

```golang
type TelemetryCreateSettings struct {
	// Logger that the factory can use during creation and can pass to the created
	// component to be used later as well.
	Logger *zap.Logger

	// TracerProvider that the factory can pass to other instrumented third-party libraries.
	TracerProvider trace.TracerProvider

	// MeterProvider that the factory can pass to other instrumented third-party libraries.
	MeterProvider metric.MeterProvider
}

// ReceiverCreateSettings configures Receiver creators.
type ReceiverCreateSettings struct {
	TelemetryCreateSettings

	// BuildInfo can be used by components for informational purposes.
	BuildInfo BuildInfo
}
```

Reply:
I like this plan, since TelemetryCreateSettings will be shareable across more than just receivers. Happy to go this route.

Reply:
First part of this in https://github.com/open-telemetry/opentelemetry-collector/pull/3984